File: src/net/machinemuse/numina/network/MusePacketNameChangeRequest.scala
package net.machinemuse.numina.network

import java.io.DataInputStream
import cpw.mods.fml.common.network.Player
import net.minecraft.client.entity.EntityClientPlayerMP
import net.machinemuse.numina.command.{NicknameMap, CommandNick}
import cpw.mods.fml.relauncher.{SideOnly, Side}
import net.minecraft.command.PlayerSelector
import net.minecraft.entity.player.EntityPlayer

/**
 * Author: MachineMuse (Claire Semple)
 * Created: 8:50 PM, 9/6/13
 */
object MusePacketNameChangeRequest extends MusePackager {
  def read(d: DataInputStream, p: Player) = {
    val username = readString(d)
    val newnick = readString(d)
    new MusePacketNameChangeRequest(p, username, newnick)
  }
}

class MusePacketNameChangeRequest(player: Player, username: String, newnick: String) extends MusePacket(player) {
  override val packager = MusePacketNameChangeRequest

  override def write() {
    writeString(username)
    writeString(newnick)
  }

  @SideOnly(Side.CLIENT)
  override def handleClient(player: EntityClientPlayerMP) {
    NicknameMap.removeName(username)
    NicknameMap.putName(username, newnick)
    val target = Option(PlayerSelector.matchOnePlayer(player, username))
    target.map(t=> t.refreshDisplayName())
  }
}

File: src/main/java/net/machinemuse/numina/network/MusePacketNameChangeRequest.java
package net.machinemuse.numina.network;

import cpw.mods.fml.relauncher.Side;
import cpw.mods.fml.relauncher.SideOnly;
import net.minecraft.client.entity.EntityClientPlayerMP;
import net.minecraft.entity.player.EntityPlayer;

import java.io.DataInputStream;

/**
 * Author: MachineMuse (Claire Semple)
 * Created: 8:50 PM, 9/6/13
 *
 * Ported to Java by lehjr on 11/14/16.
 */
public class MusePacketNameChangeRequest extends MusePacket {
    String username;
    String newnick;
    int entityID;

    public MusePacketNameChangeRequest(EntityPlayer player, String username, String newnick, int entityID) {
        this.username = username;
        this.newnick = newnick;
        this.entityID = entityID;
    }

    @Override
    public IMusePackager packager() {
        return getPackagerInstance();
    }

    @Override
    public void write() {
        writeString(username);
        writeString(newnick);
    }

    @SideOnly(Side.CLIENT)
    @Override
    public void handleClient(EntityClientPlayerMP player) {
        EntityPlayer anotherPlayer = (EntityPlayer) player.worldObj.getEntityByID(entityID);
        anotherPlayer.refreshDisplayName();
    }

    private static MusePacketNameChangeRequestPackager PACKAGERINSTANCE;

    public static MusePacketNameChangeRequestPackager getPackagerInstance() {
        if (PACKAGERINSTANCE == null)
            PACKAGERINSTANCE = new MusePacketNameChangeRequestPackager();
        return PACKAGERINSTANCE;
    }

    public static class MusePacketNameChangeRequestPackager extends MusePackager {
        @Override
        public MusePacket read(DataInputStream datain, EntityPlayer player) {
            String username = readString(datain);
            String newnick = readString(datain);
            return new MusePacketNameChangeRequest(player, username, newnick, 0);
        }
    }
}






